vm/src/vm/runners/cairo_pie.rs

```diff
@@ -134,65 +133,187 @@ impl CairoPie {
         zip_writer.finish()?;
         Ok(())
     }
+
+    #[cfg(feature = "std")]
+    pub fn read_zip_file(file_path: &Path) -> Result<CairoPie, std::io::Error> {
+        use std::io::Read;
+        use zip::ZipArchive;
+
+        let file = File::open(file_path)?;
+        let mut zip_reader = ZipArchive::new(file)?;
+
+        let mut version = vec![];
+        zip_reader
+            .by_name("version.json")?
+            .read_to_end(&mut version)?;
+        let version: CairoPieVersion = serde_json::from_slice(&version)?;
+
+        let mut metadata = vec![];
+        zip_reader
+            .by_name("metadata.json")?
+            .read_to_end(&mut metadata)?;
+        let metadata: CairoPieMetadata = serde_json::from_slice(&metadata)?;
+
+        let mut memory = vec![];
+        zip_reader.by_name("memory.bin")?.read_to_end(&mut memory)?;
+        let memory = CairoPieMemory::from_bytes(&memory)
+            .ok_or_else(|| std::io::Error::from(std::io::ErrorKind::InvalidData))?;
+
+        let mut execution_resources = vec![];
+        zip_reader
+            .by_name("execution_resources.json")?
+            .read_to_end(&mut execution_resources)?;
+        let execution_resources: ExecutionResources = serde_json::from_slice(&execution_resources)?;
+
+        let mut additional_data = vec![];
+        zip_reader
+            .by_name("additional_data.json")?
+            .read_to_end(&mut additional_data)?;
+        let additional_data: CairoPieAdditionalData = serde_json::from_slice(&additional_data)?;
+
+        Ok(CairoPie {
+            metadata,
+            memory,
+            execution_resources,
+            additional_data,
+            version,
+        })
+    }
```

Comment:
This could have a big memory peak from all the `read_to_end` calls. Consider deserializing from the readers directly, maybe with a `BufReader`.

Reply:
Alternatively, use and clear a single buffer so at least we don't have all the data in temporaries at the same time.